File: Difficulty: Easy/Chocolate Distribution Problem/chocolate-distribution-problem.cpp
class Solution {
  public:
    int findMinDiff(vector<int>& a, int m) {
        // code here
    int n = a.size();
    int ans = INT_MAX;
    sort(a.begin(),a.end());
    for(int i=0;i<=n-m;i++){
          int minval = a[i];
          int maxval = a[i+m-1];
         ans = min(ans, maxval-minval);
    }
     return ans;
    }
};

File: Difficulty: Easy/Chocolate Distribution Problem/chocolate-distribution-problem.java
class Solution {
    public int findMinDiff(ArrayList<Integer> arr, int m) {
        int res=Integer.MAX_VALUE;
         Collections.sort(arr);
         for(int i=0;i<arr.size()-m+1;i++){
             int min=arr.get(i);
             int max=arr.get(i+m-1);
             res=Math.min(res,max-min);
         }
         return res;
        
    }
}